src/README.md

```diff
@@ -0,0 +1,753 @@
+another `Isolate`.
+
+Garbage collection, as well as other operations that affect the entire heap,
+happen on a per-`Isolate` basis.
+
+Typical ways of accessing the current `Isolate` in the Node.js code are:
+
+* Given a `FunctionCallbackInfo` for a [binding function][],
+  using `args.GetIsolate()`.
+* Given a [`Context`][], using `context->GetIsolate()`.
+* Given a [`Environment`][], using `env->isolate()`.
+
+<a id="js-handles"></a>
+### JavaScript value handles
+
+All JavaScript values are accessed through the V8 API through so-called handles,
+of which there are two types: [`Local`][]s and [`Global`][]s.
+
+<a id="local-handles"></a>
+#### `Local` handles
+
+A `v8::Local` handle is a temporary pointer to a JavaScript object, where
+“temporary” usually means that is no longer needed after the current function
+is done executing. `Local` handles can only be allocated on the C++ stack.
+
+Most of the V8 API uses `Local` handles to work with JavaScript values or return
+them from functions.
+
+Whenever a `Local` handle is created, a `v8::HandleScope` or
+`v8::EscapableHandleScope` object must exist on the stack. The `Local` is then
+added to that scope and deleted along with it.
+
+When inside a [binding function][], a `HandleScope` already exists outside of
+it, so there is no need to explicitly create one.
+
+`EscapableHandleScope`s can be used to allow a single `Local` handle to be
+passed to the outer scope. This is useful when a function returns a `Local`.
+
+The following JavaScript and C++ functions are mostly equivalent:
+
+```js
+function getFoo(obj) {
+  return obj.foo;
+}
+```
+
+```c++
+v8::Local<v8::Value> GetFoo(v8::Local<v8::Context> context,
+                            v8::Local<Object> obj) {
+  v8::Isolate* isolate = context->GetIsolate();
+  v8::EscapableHandleScope handle_scope(isolate);
+
+  // The 'foo_string' handle cannot be returned from this function because
+  // it is not “escaped” with `.Escape()`.
+  v8::Local<v8::String> foo_string =
+      v8::String::NewFromUtf8(isolate,
+                              "foo",
+                              v8::NewStringType::kNormal).ToLocalChecked();
+
+  v8::Local<Value> return_value;
+  if (obj->Get(context, foo_string).ToLocal(&return_value)) {
+    return handle_scope.Escape(return_value);
+  } else {
+    // There was a JS exception! Handle it somehow.
+    return v8::Local<v8::Value>();
+  }
+}
+```
+
+See [exception handling][] for more information about the usage of `.To()`,
+`.ToLocalChecked()`, `v8::Maybe` and `v8::MaybeLocal` usage.
+
+##### Casting local handles
+
+If it is known that a `Local<Value>` refers to a more specific type, it can
+be cast to that type using `.As<...>()`:
+
+```c++
+v8::Local<v8::Value> some_value;
+CHECK(some_value->IsUint8Array());
+v8::Local<v8::Uint8Array> as_uint8 = some_value.As<v8::Uint8Array>();
+```
+
+Generally, using `val.As<v8::X>()` is only valid if `val->IsX()` is true, and
+failing to follow that rule may lead to crashes.
+
```

Comment:
could we link to a document / class here that describes common v8 types?

Reply:
I’ve added a paragraph on classes provided by V8, but what would you link here if not v8.h?

Reply:
ok, this paragraph is enough I guess!